rDNA/R/rDNA.R

```diff
@@ -1806,4 +1807,289 @@ print.dna_barplot <- function(x, trim = 30, attr = TRUE, ...) {
   x2$Entity <- sapply(x2$Entity, function(e) if (nchar(e) > trim) paste0(substr(e, 1, trim - 1), "*") else e)
   class(x2) <- "data.frame"
   print(x2)
+}
+
+#' Plot dna_barplot object.
+#'
+#' Plot barplots generated from dna_barplot.
+#'
+#' This function plots agreement and disagreement towards DNA Statements for
+#' different entities such as, e.g., "concept", "organization" or "person".
+#' 
+#' @param object A dna_barplot object.
+#' @param lab.pos,lab.neg Names for (dis-)agreement labels.
+#' @param lab Should (dis-)agreement labels and title be displayed?
+#' @param colors If \code{TRUE}, attribute colors will be used to fill the
+#'   bars.
+#' @param fontSize Text size in pts.
+#' @param barWidth Thickness of the bars. Bars will touch when set to \code{1}.
+#'   When set to \code{0.5}, space between two bars is the same as thickness of
+#'   bars.
+#' @param axisWidth Thickness of the x-axis which separates agreement from
+#'   disagreement.
+#' @param truncate Sets the number of characters to which axis labels should be truncated.
+#' @param exclude.min Reduces the plot to entities with a minimum frequency of statements.
+#'
+#' @examples
+#' \dontrun{
+#' dna_init()
+#' dna_sample()
+#'
+#' dna_openDatabase("sample.dna", coderId = 1, coderPassword = "sample")
+#' 
+#' # compute barplot data
+#' b <- dna_barplot(statementType = "DNA Statement",
+#'                  variable = "concept",
+#'                  qualifier = "agreement")
+#' 
+#' # plot barplot with ggplot2                
+#' autoplot(b)
+#' }
+#'
+#' @author Johannes B. Gruber, Tim Henrichsen
+#'
+#' @seealso \link{dna_barplot}
+#' 
+#' @importFrom ggplot2 autoplot
+#' @importFrom ggplot2 ggplot
+#' @importFrom ggplot2 aes_string
+#' @importFrom ggplot2 geom_line
+#' @importFrom ggplot2 theme_minimal
+#' @importFrom ggplot2 theme
+#' @importFrom ggplot2 geom_bar
+#' @importFrom ggplot2 position_stack
+#' @importFrom ggplot2 coord_flip
+#' @importFrom ggplot2 element_blank
+#' @importFrom ggplot2 element_text
+#' @importFrom ggplot2 scale_color_identity
+#' @importFrom ggplot2 scale_fill_identity
+#' @importFrom ggplot2 geom_text
+#' @importFrom ggplot2 .pt
+#' @importFrom ggplot2 annotate
+#' @importFrom ggplot2 scale_x_discrete
+#' @export
+autoplot.dna_barplot <- function(object, 
+                                 lab.pos = "Agreement",
+                                 lab.neg = "Disagreement",
+                                 lab = TRUE,
+                                 colors = FALSE,
+                                 fontSize = 12,
+                                 barWidth = 0.6,
+                                 axisWidth = 1.5,
```

Comment:
Not sure if I want anything changed here, but generally I'd like to give the user minimal restrictions in designing a plot. Would the user be able to override the font size using `ggplot2` after creating a plot? That would be a good thing. If it's easy to do so for the user and the default font size already looks good, is it worth specifying the font size explicitly, or would it make more sense to put an example of how to change the font size easily using `ggplot2` to the examples section of  the help file? I'm not sure what the answers to my above questions are, so not sure if anything needs to be changed. Same thing for any other specific values -- I always get suspicious when we try to micromanage the user too much. Just some general food for thought.

Reply:
Good point! I don't think it would necessarily make sense here but I'll keep your argument in mind for later functions.